### PR TITLE
Add lsof to the user image

### PR DESCRIPTION
**Pull request by yuvipanda**
lsof is helpful in figuring out ports in use, which will become
perhaps increasingly useful thanks to usage of
https://github.com/jupyterhub/jupyter-server-proxy/. There isn't
any extra security exposure here.

Fixes https://phabricator.wikimedia.org/T283984

### CODE REVIEW

**Comment by crookedstorm** [comment]
Any additional information gathering software is something I presume has security implications in fingerprinting, but I mean lsof doesn't worry me. That said, I have way more concerns about jupyter-server-proxy.  We haven't implemented network policies yet on that cluster that prevent this, but my impression here is that the proxy process runs inside the user pod? I feel like I've missed something since my attention has been elsewhere. 

I think this PR is fine, and I'll approve it. That said, I want to review the network pathway for some of the new stuff that has been added to make sure it is reasonably secure (not for the pod but for the cluster...if you get your pod or temporary proxy crashed, that's a different matter), and I also want to make sure future network hardening or changes don't remove functionality we want. I'm just commenting that here because I feel like I've been out of the loop on some of the recent things, and the expansion of functionality is quick enough to be concerning to me. If I don't have time to kick it around myself much or attack it for fun, I may hit you up sometime soon to ELI5 it for me @yuvipanda 😄

**Review by crookedstorm — DISMISSED** [review]
Seems like a fine idea. My comments are kind of tangential to this change.

**Comment by yuvipanda** [comment]
Happy to chat about jupyter-server-proxy, @crookedstorm! It doesn't add any new network flows to the infrastructure that weren't already possible, just makes it a bit easier.

**Review by dipietrom — APPROVED** [review]
